src/roads.cpp

```diff
@@ -8,13 +8,411 @@
 #include <osg/MatrixTransform>
 #include <osg/ShapeDrawable>
 
+#include <osg/Geometry>
+#include <osg/Geode>
+#include <osg/Texture2D>
+#include <osg/Program>
+#include <osg/Shader>
+#include <osg/Material>
+#include <osgSim/ShapeAttribute>
+
 #include <iostream>
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <cmath>
 
 #include "common.h"
 
 using namespace osg;
 
-osg::Node* process_roads(osg::Matrixd& ltw, const std::string & file_path)
+static const char* vertSource = R"(
+    #version 120
```

Comment:
```suggestion
    #version 420 compatibility
```
newest GLSL version with compatilibity with the older versions